rpc/eth_api.go

```diff
@@ -121,22 +122,22 @@ func (e *PublicEthAPI) GetStorageAt(address common.Address, key string, blockNum
 		return nil, err
 	}
 
-	var out types.QueryResStorage
+	var out []byte
 	e.cliCtx.Codec.MustUnmarshalJSON(res, &out)
-	return out.Value[:], nil
+	return out, nil
 }
 
 // GetTransactionCount returns the number of transactions at the given address up to the given block number.
-func (e *PublicEthAPI) GetTransactionCount(address common.Address, blockNum rpc.BlockNumber) (hexutil.Uint64, error) {
+func (e *PublicEthAPI) GetTransactionCount(address common.Address, blockNum rpc.BlockNumber) (*hexutil.Uint64, error) {
 	ctx := e.cliCtx.WithHeight(blockNum.Int64())
 	res, _, err := ctx.QueryWithData(fmt.Sprintf("custom/%s/nonce/%s", types.ModuleName, address), nil)
 	if err != nil {
-		return 0, err
+		return nil, err
 	}
 
-	var out types.QueryResNonce
+	var out *hexutil.Uint64
```

Comment:
Reason for unmarshalling a pointer to this pointer? If the type that is being marshalled is just the hexutil uint64 type, is this necessary and confirmed to be functional?

Reply:
Following whats here: https://github.com/ethereum/go-ethereum/blob/b8a94571397c1afd5579f1e319aff05a676c917d/internal/ethapi/api.go#L1227

Reply:
I'm more referring to the [line below](https://github.com/ChainSafe/ethermint/pull/94/files/9812df303517a9324710585cbabbba196f066bd7#diff-97f02ad1e0a5dd042d5a74fc21e307b9R139), since it is unmarshalling into a pointer to that hexutil pointer, it may work, but just wanted to confirm it unmarshalls correctly if the query returns a non-zero value.